app/routers/admin_router.py

```
@@ -247,11 +250,33 @@ async def create_post(
+    # --- START PIN PRIORITY VALIDATION AND ASSIGNMENT ---
+    pin_priority_value = None
+    if pin_priority and pin_priority.strip():
+        try:
+            priority_int = int(pin_priority)
+            if priority_int < 0:
+                raise ValueError("Priority cannot be negative.")
+            pin_priority_value = priority_int
+        except ValueError:
+            # Handle error and reload form, preserving existing inputs (simplified error handling)
+            return request.state.templates.TemplateResponse(
+                 "admin/post_edit.html",
```

Flake8 (7.3.0)

[error] continuation line over-indented for hanging indent

(E126)